src/Text/Pandoc/App/CommandLineOptions.hs

```diff
@@ -536,8 +536,9 @@ options =
                  "" -- "Make slide shows include all the needed js and css (deprecated)"
 
     , Option "" ["embed-resources"]
-                 (NoArg
-                  (\opt -> return opt { optEmbedResources = True }))
+                 (OptArg
+                  (\arg opt -> return opt { optEmbedResources = maybe True (\argStr -> argStr == "true") arg })
```

Comment:
This just checks for the string `"true"`.  So, `--embed-resources=1`, `--embed-resources=yes`, `--embed-resources=True`, etc. will all be *silently* interpreted as `--embed-resources=false`.

I think a better behavior would raise an error if the optional argument is present and is not `"true"` or `"false"`.

For examples of this kind of checking and error raising, see e.g. the code for the option `eol`.

Reply:
I also think that, either in this PR or in a later change, we should uniformly change all boolean options so they take this optional parameter.  So the validation code should be factored out into a separate function that can be used elsewhere.

Reply:
I've update the PR to include the suggested changes.